odinmain/src/main/kotlin/me/odinmain/features/impl/dungeon/BloodCamp.kt

```diff
@@ -205,11 +206,17 @@ object BloodCamp : Module(
                 currVector.zCoord + speedVectors.zCoord * ping
             )
 
-            val boxOffset: Vec3 = Vec3((boxSize/2).unaryMinus(),1.5,(boxSize/2).unaryMinus())
-            val pingAABB = AxisAlignedBB(boxSize,boxSize,boxSize, 0.0, 0.0, 0.0).offset(boxOffset).offset(pingPoint)
-            val endAABB = AxisAlignedBB(boxSize,boxSize,boxSize, 0.0, 0.0, 0.0).offset(boxOffset).offset(endPoint)
+            val renderEndPoint = getRenderVector(endPoint, event.partialTicks, renderData.lastEndPoint)
+            val renderPingPoint = getRenderVector(pingPoint, event.partialTicks, renderData.lastPingPoint)
 
-            val time = data.time ?: return@forEach
+            renderData.lastEndPoint = endPoint
+            renderData.lastPingPoint = pingPoint
+
+            val boxOffset = Vec3((boxSize/2).unaryMinus(),1.5,(boxSize/2).unaryMinus())
+            val pingAABB = AxisAlignedBB(boxSize,boxSize,boxSize, 0.0, 0.0, 0.0).offset(boxOffset).offset(renderPingPoint)
+            val endAABB = AxisAlignedBB(boxSize,boxSize,boxSize, 0.0, 0.0, 0.0).offset(boxOffset).offset(renderEndPoint)
```

Comment:
you should be able to just do .offset(boxOffset + renderEndPoint), there is an operator function in vecutils for it